### PR TITLE
[jest-image-snapshot] Fix customDiffConfig type

### PR DESCRIPTION
**Pull request by varholak-peter**
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:

`customDiffConfig` should accept partial `SSIMOptions` as that is used as an argument here: 

**jest-image-snapshot usage**
[jest-image-snapshot/src/diff-snapshot.js](https://github.com/americanexpress/jest-image-snapshot/blob/d60e2e672e11dd9668122bcf8c3ac8a36bed0a8e/src/diff-snapshot.js#L102)

**ssim function**
[ssim/src/index.ts](https://github.com/obartra/ssim/blob/master/src/index.ts#L95)

### CODE REVIEW

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@varholak-peter thx!
fyi: as there is no test change, this one will still wait for additional review by DT maintainer

**Comment by varholak-peter** [comment]
No problem, to be honest I am unsure how to test the `Partial<T>` type. ¯\_(ツ)_/¯

**Comment by peterblazejewicz** [comment]
any way would be OK, e.g.:
```ts
import { PixelmatchOptions } from 'pixelmatch';
import { Options as SSIMOptions } from 'ssim.js';

const ssimConfig: Partial<SSIMOptions> = {
    ssim: 'bezkrovny',
    windowSize: 1024,
};

const pixelmatchOptions: PixelmatchOptions = {
    includeAA: true,
    threshold: 0.2,
};

configureToMatchImageSnapshot({
    customDiffConfig: ssimConfig,
});
configureToMatchImageSnapshot({
    customDiffConfig: pixelmatchOptions,
});
```
btw: I'm always poiting for DT owners, that for users, like us, adding that requirement to type 'Partial<Something>` when consuming types, is a PITA, just wrong devex

**Review by andrewbranch — APPROVED** [review]

**Comment by varholak-peter** [comment]
Ready to merge

**Pull request merged**
